habushu-maven-plugin/src/test/resources/specifications/containerize-dependencies.feature

```diff
@@ -9,4 +9,11 @@ Feature: Test staging source files of specified dependencies and transitive path
   Scenario: No Habushu-type dependencies are specified with the default configuration
     Given no Habushu-type dependencies
     When the containerize-dependencies goal is executed
-    Then no source files are staged in the build directory
\ No newline at end of file
+    Then no source files are staged in the build directory
+
+  Scenario: One Habushu-type dependency is specified with Dockerfile updates enabled
+    Given a single dependency with packaging type habushu
+     And a dockerfile to update
+    When the containerize-dependencies goal is executed
+    Then all the source files to build the dependency are staged in the build directory
+     And Dockerfile is updated to build a virtual environment for the dependency
```

Comment:
Have we verified that the resulting docker image is configured to use the resulting `venv` correctly? Based on my testing, when I run the build and run a docker image with the generated habushu code, my `which python` isn't pointing to the `venv` so it doesn't get used.

Reply:
Could just be an issue with my underlying image though.

Reply:
For me, I am using `FROM python:3.11-slim as final` as my final image, and I had to add the following lines to point python to the venv:
```
RUN unlink /venv/bin/python
RUN ln -s /usr/local/bin/python /venv/bin/python
```
Will probably vary by final image though.

Reply:
Could make this generic to link the venv to the underlying python executable by including the following in `#HABUSHU_FINAL_STAGE`:
```
RUN unlink /venv/bin/python
RUN ln -s $(which python) /venv/bin/python
```
This will delete the potentially incorrect symlink and create a correct one base on the location of the final image python executable.